src/stores/store/secrets/hashicorp/hashicorp.go

```diff
@@ -134,16 +138,19 @@ func (s *Store) List(_ context.Context) ([]string, error) {
 	return keysStr, nil
 }
 
-func (s *Store) Delete(_ context.Context, id, version string) error {
-	if version == "" {
-		err := errors.NotSupportedError("secret version must be specified to delete")
-		s.logger.Warn(err.Error())
-		return err
+func (s *Store) Delete(ctx context.Context, id string) error {
+	logger := s.logger.With("id", id)
+
+	versionList, err := s.db.ListVersions(ctx, id, false)
```

Comment:
this block can be moved to a private function given that it logs so you can return the err directly and avoid duplicating the errMessage definition. Linting should fail anyways.

Reply:
You are right, I will encapsulate this as private method